### PR TITLE
fix(widgets/image): avoid init NULL obj

### PR DESCRIPTION
**Pull request by Sunrisepeak**
### Description of the feature or fix

avoid to init NULL obj, and return to user side to process when obj is NULL 
(user side to release some memory or do something)

---

avoid potential dereference null pointer

https://github.com/lvgl/lvgl/blob/f36890102fbbda2d1edddd9006c08d787f0acca8/src/widgets/image/lv_image.c#L126-L132

https://github.com/lvgl/lvgl/blob/f36890102fbbda2d1edddd9006c08d787f0acca8/src/core/lv_obj_class.c#L45-L51

https://github.com/lvgl/lvgl/blob/f36890102fbbda2d1edddd9006c08d787f0acca8/src/core/lv_obj_class.c#L103-L106

https://github.com/lvgl/lvgl/blob/f36890102fbbda2d1edddd9006c08d787f0acca8/src/core/lv_obj_pos.c#L276-L279

### Notes
- Update the [Documentation](https://github.com/lvgl/lvgl/tree/master/docs) if needed.
- Add [Examples](https://github.com/lvgl/lvgl/tree/master/examples) if relevant.
- Add [Tests](https://github.com/lvgl/lvgl/blob/master/tests/README.md) if applicable.
- If you added new options to `lv_conf_template.h` run [lv_conf_internal_gen.py](https://github.com/lvgl/lvgl/blob/master/scripts/lv_conf_internal_gen.py) and update [Kconfig](https://github.com/lvgl/lvgl/blob/master/Kconfig).
- Run `scripts/code-format.py` ([astyle](http://astyle.sourceforge.net/install.html) version [v3.4.12](https://github.com/szepeviktor/astyle/releases/tag/v3.4.12) needs to be installed) and follow the [Code Conventions](https://docs.lvgl.io/master/CODING_STYLE.html).
- Mark the Pull request as [Draft](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-stage-of-a-pull-request) while you are working on the first version, and mark is as _Ready_ when it's ready for review.
- When changes were requested, [re-request review](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/requesting-a-pull-request-review) to notify the maintainers.
- Help us to review this Pull Request! Anyone can [approve or request changes](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/approving-a-pull-request-with-required-reviews).

### CODE REVIEW

File: src/widgets/image/lv_image.c [hunk 1]
```diff
@@ -128,7 +128,9 @@ lv_obj_t * lv_image_create(lv_obj_t * parent)
 {
     LV_LOG_INFO("begin");
     lv_obj_t * obj = lv_obj_class_create_obj(MY_CLASS, parent);
-    lv_obj_class_init_obj(obj);
```
Comment (XuNeo) [inline src/widgets/image/lv_image.c]: How about check NULL obj in `lv_obj_class_init_obj` and return directly?

Comment (Sunrisepeak) [reply src/widgets/image/lv_image.c]: check in `lv_obj_class_init_obj` is better, I have updated the patch

**Comment by kisvegabor** [comment]
We are updating the MicroPython CI and the fail is probably related to it. Re-running now.
cc @liamHowatt

**Comment by liamHowatt** [comment]
If you rebase it should pass now. Sorry for the inconvenience.

**Comment by Sunrisepeak** [comment]
> If you rebase it should pass now. Sorry for the inconvenience.

done

**Review by XuNeo — APPROVED** [review]
LGTM.

**Review by liamHowatt — APPROVED** [review]

**Comment by FASTSHIFT** [comment]
@kisvegabor @liamHowatt 
Sorry, I forgot to select squash by mistake :(

**Comment by kisvegabor** [comment]
No problem, I've rebased and force pushed to `master`. I haven't done that for a long time so I hope all will be good :crossed_fingers:

**Pull request merged**